pkg/controller/storagecluster/stork.go

```diff
@@ -883,6 +883,7 @@ func getStorkSchedDeploymentSpec(
 ) *apps.Deployment {
 	pullPolicy := imagePullPolicy(cluster)
 	replicas := int32(3)
+	maxSurge := intstr.FromInt(0)
```

Comment:
don't know why we have a 0 here? 
also this
```
maxUnavailable declares the maximum number (or percentage) of unavailable pods during the update. If maxSurge is set to 0, this field cannot be 0.
```

Reply:
The default value of maxUnavailable is 25%. So we can set maxSurge to 0 here.

Reply:
Max surge = 0 means that without adding an extra pod, the deployment can start terminating pods during an upgrade.

Anyway, I code to make it similar to pvc controller. The problem was because of maxUnavailable. The default value of 25% was preventing even 1/3 pods to go down.

Reply:
oh we already have this update strategy in pvc controller, just saw that.